File: devorbit-api/src/main/java/vn/edu/uit/devorbit_api/dto/student/GroupPlanResponse.java
package vn.edu.uit.devorbit_api.dto.student;

import vn.edu.uit.devorbit_api.entity.GroupPlan;

import java.time.LocalDate;

public record GroupPlanResponse(
    Long id,
    String title,
    String description,
    String creatorStudentCode,
    LocalDate deadline,
    boolean active,
    String createdAt,
    boolean deleteRequested,
    String deleteRequestedBy
) {
    public static GroupPlanResponse from(GroupPlan gp) {
        return new GroupPlanResponse(
            gp.getId(),
            gp.getTitle(),
            gp.getDescription(),
            gp.getCreatorStudentCode(),
            gp.getDeadline(),
            gp.isActive(),
            gp.getCreatedAt() != null ? gp.getCreatedAt().toString() : null,
            gp.isDeleteRequested(),
            gp.getDeleteRequestedBy()
        );
    }
}

File: devorbit-mobile/app/src/main/java/vn/edu/uit/devorbit/mobile/data/remote/dto/GroupPlanResponse.kt
package vn.edu.uit.devorbit.mobile.data.remote.dto

data class GroupPlanResponse(
    val id: Long,
    val title: String,
    val description: String?,
    val creatorStudentCode: String,
    val deadline: String?,
    val active: Boolean,
    val createdAt: String?,
    val deleteRequested: Boolean = false,
    val deleteRequestedBy: String? = null
)
